public/modules/custom/helfi_rekry_content/src/EventSubscriber/ImageImportSubscriber.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Drupal\helfi_rekry_content\EventSubscriber;
+
+use Drupal\Core\File\FileSystem;
+use Drupal\media\Entity\Media;
+use Drupal\migrate\Event\MigrateEvents;
+use Drupal\migrate\Event\MigrateImportEvent;
+use Drupal\migrate\Event\MigratePostRowSaveEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class for subscribing to image import events.
+ */
+class ImageImportSubscriber implements EventSubscriberInterface {
+  /**
+   * URI for job listing images directory.
+   */
+  protected const IMAGES_DIR = 'public://job_listing_images/';
+
+  /**
+   * Constructs a new instance.
+   *
+   * @param \Drupal\Core\File\FileSystem $fileSystem
+   *   The file system service.
+   */
+  public function __construct(private FileSystem $fileSystem) {
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents(): array {
+    return [
+      MigrateEvents::POST_ROW_SAVE => 'postRowSave',
+      MigrateEvents::PRE_IMPORT => 'preImport',
+    ];
+  }
+
+  /**
+   * Create or update a media Entity of the file getting uploaded.
+   *
+   * @param \Drupal\migrate\Event\MigratePostRowSaveEvent $event
+   *   The event object.
+   */
+  public function postRowSave(MigratePostRowSaveEvent $event): void {
+    if (in_array($event->getMigration()->id(), $this->getImageMigrations())) {
```

Comment:
This condition should probably be reversed and return early to avoid unnecessary nested conditions. Like
```php
if (!in_array($event->getMigration()->id(), $this->getImageMigrations())) {
  return;
}
```